2003-02-10  Jakub Jelinek  <jakub@redhat.com>

	* config/tc-s390.c (md_pseudo_table): Add .file and .loc.
	* config/tc-s390.h (DWARF2_LINE_MIN_INSN_LENGTH): Set to 2.

--- gas/config/tc-s390.c.jj	2003-02-05 18:24:45.000000000 -0500
+++ gas/config/tc-s390.c	2003-02-10 07:56:19.000000000 -0500
@@ -82,7 +82,7 @@ static void s390_literals PARAMS ((int))
 
 const pseudo_typeS md_pseudo_table[] =
 {
-  { "align", s_align_bytes, 0 },
+  { "align",	s_align_bytes,	0 },
   /* Pseudo-ops which must be defined.  */
   { "bss",      s390_bss,       0 },
   { "insn",     s390_insn,      0 },
@@ -93,6 +93,8 @@ const pseudo_typeS md_pseudo_table[] =
   { "quad",     s390_elf_cons,  8 },
   { "ltorg",    s390_literals,  0 },
   { "string",   stringer,       2 },
+  { "file",	(void (*) PARAMS ((int))) dwarf2_directive_file, 0 },
+  { "loc",	dwarf2_directive_loc, 0 },
   { NULL,	NULL,		0 }
 };
 
--- gas/config/tc-s390.h.jj	2003-01-24 18:17:06.000000000 -0500
+++ gas/config/tc-s390.h	2003-02-10 07:54:50.000000000 -0500
@@ -1,5 +1,5 @@
 /* tc-s390.h -- Header file for tc-s390.c.
-   Copyright 2000, 2001, 2002 Free Software Foundation, Inc.
+   Copyright 2000, 2001, 2002, 2003 Free Software Foundation, Inc.
    Written by Martin Schwidefsky (schwidefsky@de.ibm.com).
 
    This file is part of GAS, the GNU Assembler.
@@ -98,3 +98,5 @@ extern long md_pcrel_from_section PARAMS
 
 extern void s390_md_end PARAMS ((void));
 #define md_end() s390_md_end ()
+
+#define DWARF2_LINE_MIN_INSN_LENGTH 2
